scipy/special/specfun/specfun.f

```diff
@@ -10007,35 +10007,44 @@ SUBROUTINE JYZO(N,NT,RJ0,RJ1,RY0,RY1)
         DIMENSION RJ0(NT),RJ1(NT),RY0(NT),RY1(NT)
         PI=3.141592653589793D0
 C       -- Newton method for j_{N,L}
-C       1) initial guess for j_{N,1}
-        IF (N.LE.20) THEN
-           X=2.82141+1.15859*N
+C       initial guess for j_{N,1}
+        IF (N.EQ.0) THEN
+            X=2.4
         ELSE
-C          Abr & Stg (9.5.14)
-           X=N+1.85576*N**0.33333+1.03315/N**0.33333
+            X=N+1.85576*N**0.33333+1.03315/N**0.33333
         ENDIF
+C       iterate
         L=0
-C       2) iterate
-        XGUESS=X
 10      X0=X
         CALL JYNDD(N,X,BJN,DJN,FJN,BYN,DYN,FYN)
         X=X-BJN/DJN
-        IF (X-X0.LT.-1) X=X0-1
-        IF (X-X0.GT.1) X=X0+1
         IF (DABS(X-X0).GT.1.0D-11) GO TO 10
-C       3) initial guess for j_{N,L+1}
-        IF (L.GE.1)THEN
-           IF (X.LE.RJ0(L)+0.5) THEN
-              X=XGUESS+PI
-              XGUESS=X
-              GO TO 10
-           ENDIF
-        END IF
         L=L+1
         RJ0(L)=X
-C       XXX: should have a better initial guess for large N ~> 100 here
-        X=X+PI+MAX((0.0972d0+0.0679*N-0.000354*N**2)/L, 0d0)
+C       initial guess for j_{N,2}
+        IF (L.EQ.1) THEN
+            IF (N.EQ.0) THEN
+                X=5.52
+            ELSE
+                X=N+3.24469944*N**0.33333+3.14569854/N**0.33333
+            ENDIF
+C       initial guess for j_{N,L}
```

Comment:
Depends on how you look at it. $L$ was updated to $L+1$ a few lines above, so $L$ is technically correct, but perhaps $L+1$ is more clear?

Reply:
Ah yes, you’re right

Reply:
Wait, in the previous iteration, L is set to L+1, and then RJ0(L) is set to the computed value for the Lth zero. Now we’re getting an initial approximation for the (L+1)th zero. We will later set L to L+1 and RJ0(L) to th next computed value. 

It’s confusing to say initial guess for $ j_{N, L}$ right, because this is RJ0(L) and if we don’t have a value for it yet, how is it appearing on the right hand side? Or have I made an off by one error in my thinking?

Reply:
Ah I see now that you're right. It should be L+1.
In my mind L was incremented after assigning to RJ0, but L is incremented before...